### PR TITLE
Add 3D demo

### PR DESCRIPTION
**Pull request by yutakaoka**
This adds a 3D demo example. Please review it.

### CODE REVIEW

**Comment by yutakaoka** [comment]
I signed it!

**Review by macmason — APPROVED** [review]

File: docs/source/index.rst [hunk 1]
```diff
@@ -71,10 +71,16 @@ use ``roslaunch`` to bring up the demo:
...
+    # Download the 2D example bag.
     wget -P ~/Downloads https://storage.googleapis.com/cartographer-public-data/bags/toyota_hsr/cartographer_toyota_hsr_demo.bag
 
-    # Launch the demo.
-    roslaunch cartographer_toyota_hsr demo_hsr.launch bag_filename:=${HOME}/Downloads/cartographer_toyota_hsr_demo.bag
+    # Launch the 2D demo.
+    roslaunch cartographer_toyota_hsr demo_hsr_2d.launch bag_filename:=${HOME}/Downloads/cartographer_toyota_hsr_demo.bag
+
+    # Download the 3D example bag.
+    wget -P ~/Downloads https://s3-us-west-1.amazonaws.com/cartographer-hsr/cartographer_hsr_example_3d.bag
```
Comment (damonkohler) [inline docs/source/index.rst]: You should recompress this with LZ4. BZ2 is too slow for 3D data.

rosbag compress --lz4 cartographer_hsr_example_3d.bag

File: cartographer_toyota_hsr/configuration_files/toyota_hsr_3d.lua [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  use_laser_scan = false,
+  use_multi_echo_laser_scan = false,
+  num_point_clouds = 1,
+  lookup_transform_timeout_sec = 0.2,
+  submap_publish_period_sec = 0.3,
+  pose_publish_period_sec = 5e-3,
+}
+
+MAP_BUILDER.use_trajectory_builder_3d = true
+MAP_BUILDER.sparse_pose_graph.optimization_problem.huber_scale = 5e3
```
Comment (damonkohler) [inline cartographer_toyota_hsr/configuration_files/toyota_hsr_3d.lua]: Did you try the backpack_3d.lua settings? They seem to work well.

I'm not suggesting you load them instead, that won't work obviously. But you may want to copy the MAP_BUILDER configuration from there.

**Review by damonkohler — CHANGES_REQUESTED** [review]
Ideally, the VLP point clouds should be partial revolutions (e.g. one per UDP packet). That allows SLAM to unwarp them.

In a separate PR, can you add the URDF?

File: docs/source/index.rst (hunk 1 above)
Comment (macmason) [reply docs/source/index.rst]: #til, rosbag has lz4 support.

Comment (macmason) [reply docs/source/index.rst]: @yutakaoka I've recompressed and re-uploaded the bag. The filename is the same, but you should double-check.

@damonkohler if you'd like to host this one, send a followup PR?

Comment (damonkohler) [reply docs/source/index.rst]: I've uploaded the lz4 compressed bag here:

https://storage.googleapis.com/cartographer-public-data/bags/toyota_hsr/cartographer_toyota_hsr_demo_3d.bag

File: cartographer_toyota_hsr/configuration_files/toyota_hsr_3d.lua (hunk 2 above)
Comment (yutakaoka) [reply cartographer_toyota_hsr/configuration_files/toyota_hsr_3d.lua]: I tried that. Both of them looked fine, but I applied your suggestion.

File: docs/source/index.rst (hunk 1 above)
Comment (yutakaoka) [reply docs/source/index.rst]: Thanks, I fixed it.

**Comment by yutakaoka** [comment]
Regarding the URDF, let me confirm if I can do that.

**Review by damonkohler — APPROVED** [review]

**Pull request merged**